dali/operators/reader/numpy_reader_op.cc

```diff
@@ -14,15 +14,107 @@
 
 #include <string>
 
+#include "dali/pipeline/data/views.h"
 #include "dali/kernels/common/transpose.h"
 #include "dali/core/static_switch.h"
 #include "dali/operators/reader/numpy_reader_op.h"
 
 namespace dali {
 
-#define NUMPY_ALLOWED_TYPES \
-  (bool, uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, float16, \
-  double)
+TensorListShape<> NumpyReader::GetSliceArg(ArgumentWorkspace &ws, const char *name) {
+  if (spec_.HasArgument(name)) {
+    return uniform_list_shape(batch_size_, spec_.GetRepeatedArgument<int>(name));
+  } else if (spec_.HasTensorArgument(name)) {
```

Comment:
I don't think it is going  to work. Because when you run prefetching you can read 1000 samples ahead while the rest of the pipeline hasn't done any iterations, so you won't get right values of shape and anchor for the constitutive batches.

Reply:
Even when the shuffling is off, you read ahead more than a batch size of data.

Reply:
yes, I think that is an issue. Do you have an idea how to fix that?

Reply:
I guess in that case, read-ahead would need to be disabled, right?

Reply:
Maybe own implementation of ReadOne? But this would kill any readahead and probably be a bottleneck, still worth to try.
Try to disable read ahead and check how much it affects the perf.

Reply:
I did that and it is still better than other solutions, such as using external source. My problem is here: in order to disable the prefetching during shuffle, I would need to know during instantiation that tensor args will be used, what I don't. So what I could do: do a run time check and if Tensor arguments are present AND shuffling is enabled, throw an error. This is an ugly solution though. Do you have a better idea?